File: 8kyu/(8 kyu) Returning Strings/(8 kyu) Returning Strings.dart
// 1 Plain solution
String greet(String name) {
  final String str1 = "Hello, ";
  final String str2 = " how are you doing today?";

  return str1 + name + str2;
}

// 2 Optimized solution
String greet(String name) {
  return "Hello, " + name + " how are you doing today?";
}

// 3 Clever solution
String greet(String name) {
  return "Hello, $name how are you doing today?";
}

// 4 Coding golf
greet(n) => "Hello, $n how are you doing today?";

File: 8kyu/(8 kyu) Returning Strings/(8 kyu) Returning Strings.py
# 1 Plain solution
def greet(name):
    str1 = "Hello, "
    str2 = " how are you doing today?"
    return str1 + name + str2


# 2 Optimized solution
def greet(name):
    return "Hello, " + name + " how are you doing today?"

# 3 Clever solution
def greet(name):
    return f"Hello, {name} how are you doing today?"

# 4 Coding golf
greet=lambda n:f"Hello, {n} how are you doing today?"